src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -310,38 +319,59 @@ private static File getUserConfigDir() {
         return dir;
     }
 
+    /* Append the osAppendix to the binPath starting at position endIdx */
+
+    private static File getInstalledConfigDir(String binPath, int endIdx, String osAppendix){
+        if (endIdx != -1){
+            binPath = binPath.substring(0, endIdx);
+        }
+        return new File(binPath, osAppendix);
+    }
+
     /**
-     * Returns default Dataloader configuration directory in user directory
+     * Returns default Dataloader installed configuration directory
      *
-     * @return
-     *  Default Dataloader configuration directory
+     * @return Default Dataloader configuration directory
      */
-    private static File getDefaultConfigDir() {
+    private static File getInstalledConfigDir() {
+
+        URI uri;
+        String path;
+        try {
+            // This return the jar's location
+            // The current recommendation (with JDK 1.7+) is to convert URL → URI → Path.
+            uri = Controller.class.getProtectionDomain().getCodeSource().getLocation().toURI();
+            path = Paths.get(uri).toFile().getAbsolutePath();
+            logger.debug("The installation binary location is: " + path);
+        } catch (URISyntaxException e) {
+            logger.error("Can find binary's location", e);
+            throw new RuntimeException(e);
+        }
+
         File dir;
         switch (OS_TYPE) {
             case WINDOWS: {
-                dir = new File(System.getProperty("user.dir"), "conf");
+                //For windows,﻿filepath is﻿C:\Program Files\salesforce.com\Data Loader\dataloader-xxx-uber.jar
+                dir = getInstalledConfigDir(path, path.lastIndexOf(File.separator),  "conf");
```

Comment:
Java provides path traversal and cracking functions. It is generally better to use those than to do it yourself using `lastIndexOf` or `indexOf` and `substring`. However, this does seem to be done correctly here.

Reply:
It is a little bit tricky for different OS here.